packages/toolpad-components/src/DataGrid.tsx

```diff
@@ -108,7 +114,14 @@ function inferColumnType(value: unknown): string {
   switch (typeof value) {
     case 'number':
     case 'boolean':
+      return valueType;
     case 'string':
+      if (IMAGE_REGEX.test(value)) {
+        return 'image';
+      }
+      if (URL_REGEX.test(value)) {
```

Comment:
Just an idea - we could also probably try to use native [`URL` construct](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) - seems that when we pass invalid url it throws an error. One benefit of that is that we wouldn't need to "own" regexp in our code:
```
try {
  new URL(value)
  return 'link'
} catch (error) {}
```

Maybe it's not needed right now, but it might be worth to think if it wouldn't be useful to have these methods defined somewhere in `utils` section, I believe we might also want to reuse this logic in the future for other components 🤔

Reply:
Always use `URL` to parse urls: https://github.com/mui/mui-toolpad/blob/8e6d211d2beb4a346a4dd153105f5190b8a842f7/packages/toolpad-app/src/utils/strings.ts#L56.When you read a url parsing regex on stackoverflow, it will most likely be wrong.

Same for testing the images, parse the `pathname` out and use `.endsWidth`.